src/plugins/registration.go

```diff
@@ -153,23 +156,43 @@ func (r *OneTimeRegistration) areDataplaneSoftwareDetailsReady() error {
 func (r *OneTimeRegistration) registerAgent() {
 	var details []*proto.NginxDetails
 
-	for _, proc := range r.processes {
-		// only need master process for registration
-		if proc.IsMaster {
-			nginxDetails := r.binary.GetNginxDetailsFromProcess(proc)
-			details = append(details, nginxDetails)
-			// Reading nginx config during registration to populate nginx fields like access/error logs, etc.
-			_, err := r.binary.ReadConfig(nginxDetails.GetConfPath(), nginxDetails.NginxId, r.env.GetSystemUUID())
-			if err != nil {
-				log.Warnf("Unable to read config for NGINX instance %s, %v", nginxDetails.NginxId, err)
+	backoffSetting := backoff.BackoffSettings{
+		InitialInterval: softwareDetailsOperationInterval,
+		MaxInterval:     softwareDetailsOperationInterval,
+		MaxElapsedTime:  nginxStartMaxWaitTime,
+		Jitter:          backoff.BACKOFF_JITTER,
+		Multiplier:      backoff.BACKOFF_MULTIPLIER,
+	}
+
+	findMaster := func() error {
+		for _, proc := range r.processes {
+			// only need master process for registration
+			if proc.IsMaster {
+				nginxDetails := r.binary.GetNginxDetailsFromProcess(proc)
+				details = append(details, nginxDetails)
+				if len(details) > 0 {
+					// Reading nginx config during registration to populate nginx fields like access/error logs, etc.
+					_, err := r.binary.ReadConfig(nginxDetails.GetConfPath(), nginxDetails.NginxId, r.env.GetSystemUUID())
+					if err != nil {
+						log.Warnf("Unable to read config for NGINX instance %s, %v", nginxDetails.NginxId, err)
+					}
+					log.Info("Master process has been found")
+					return nil
+				}
+			} else {
+				log.Tracef("NGINX non-master process: %d", proc.Pid)
+				return nil
```

Comment:
This `return` causes the whole function to exit, meaning that the function will only ever find the master process if it is the first process in `r.processes`.